src/tests/test_utils/common_test_utils/src/ov_tensor_utils.cpp

```diff
@@ -305,150 +305,150 @@ ov::Tensor create_and_fill_tensor_consistently(const ov::element::Type element_t
 
 constexpr double eps = std::numeric_limits<double>::epsilon();
 
-inline double less(double a, double b) {
+template <typename aT, typename bT>
+inline double less(aT a, bT b) {
     return std::fabs(a - b) > eps && a < b;
 }
 
 inline double less_or_equal(double a, double b) {
     bool res = true;
     if (std::isnan(a) || std::isnan(b)) {
         res = false;
+    } else if (std::isinf(b) && std::isinf(b)) {
+        res = true;
     } else if (std::isinf(b) && b > 0) {
-        // b is grater than any number or eq the +Inf
+        // b is greater than any number or eq the +Inf
         res = true;
     } else if (std::isinf(a) && a > 0) {
         res = false;
     } else {
-        res = (std::fabs(b - a) <= (std::fmax(std::fabs(a), std::fabs(b)) * eps) || a < b);
+        res = std::fabs(b - a) <= eps || a <= b;
     }
-    double eq_midle_res = std::fabs(b - a);
-    bool eq_res = (std::fabs(b - a) <= (std::fmax(std::fabs(a), std::fabs(b)) * eps));
     return res;
 }
 
-struct Error {
-    double max = 0.;
-    double mean = 0.;
-    size_t max_coordinate = 0;
-    size_t count = 0;
-    double threshold;
+template <typename T1, typename T2>
+inline bool check_values_suitabl_for_comparison(double value1, double value2) {
+    bool res = true;
+    auto max_val1 = std::numeric_limits<T1>::max();
+    auto min_val1 = std::numeric_limits<T1>::lowest();
+    auto max_val2 = std::numeric_limits<T2>::max();
+    auto min_val2 = std::numeric_limits<T2>::lowest();
+    if (std::isnan(value1) && std::isnan(value2)) {
+        res = false;
+    } else if ((std::isinf(value1) || value1 >= max_val1) && (std::isinf(value2) || value2 >= max_val2)) {
+        res = false;
+    } else if ((std::isinf(value1) || value1 <= min_val1) && std::isinf(value2) || value2 <= min_val2) {
+        res = false;
+    }
+
+    return res;
+}
+
+class Error {
+protected:
+    struct IncorrectValue {
+        size_t coordinate;
+        double actual_value, expected_value, threshold;
+
+        IncorrectValue(double in_actual_value, double in_expected_value, double in_threshold, size_t in_coordinate)
+            : actual_value(in_actual_value),
+              expected_value(in_expected_value),
+              threshold(in_threshold),
+              coordinate(in_coordinate) {}
+    };
 
-    Error(double _threshold) : threshold(_threshold) {}
+    std::vector<IncorrectValue> incorrect_values_abs;
+    std::vector<IncorrectValue> incorrect_values_rel;
+    double abs_threshold, rel_threshold;
```

Comment:
if we store only one threshold, looks like we can remove second one

Reply:
these thresholds for all errors and here it will be two different thresholds abs and rel, but we have IncorrectValue vector for each specific error and in each IncorrectValue element it will be one calculated threshold, which was calculated based on rel, abs threshold and expected values